tests/spec/fs.chmod.spec.js

```diff
@@ -71,3 +71,71 @@ describe('fs.chmod, fs.fchmod', function() {
     });
   });
 });
+
+describe('fsPromise.chmod', function() {
+  beforeEach(util.setup);
+  afterEach(util.setup);
+
+  it('should be a function', function() {
+    var fsPromise = util.fs().promises;
+    expect(typeof fsPromise.chmod).to.equal('function');
+  });
+
+  it('should return a promise', function() {
+    var fsPromise = util.fs().promises;
+    expect(fsPromise.chmod()).to.be.a('Promise');
+  });
+
+  it('should allow for updating mode of a given file', function() {
+    var fsPromise = util.fs().promises;
+
+    return fsPromise.open('/file', 'w')
```

Comment:
I'm wondering about how you've nested these `.then()` and `.catch()`... 
Is there a better way these can be nested? 
Maybe instead of doing the `return` within `return`, just call the functions and have the `.then()` and `.catch()` for that function nested separately? 
i.e
``` 
var fsPromise = util.fs().promises;
     return fsPromise.open('/file', 'w')
         .then(()=>{
              fsPromise.chmod('/file', 0o444)
                  .then(...)
                  .catch(...);
          }) 
         .catch(...);
```
I'm not sure what's considered best practice, or if it there is even a 'better' or 'worse' way to do it at all...

Reply:
I've removed the redundant ``.catch()`` blocks. 
Thanks for the review.